adapters/adnuntius/adnuntius.go

```diff
@@ -0,0 +1,311 @@
+	if gdpr != "" && consent != "" {
+		q.Set("gdpr", gdpr)
+		q.Set("consentString", consent)
+	}
+	q.Set("tzo", fmt.Sprint(tzo))
+	q.Set("format", "json")
+
+	url := a.endpoint + "?" + q.Encode()
+	return url, nil
+}
+
+/*
+	Generate the requests to Adnuntius to reduce the amount of requests going out.
+*/
+func (a *adapter) generateRequests(ortbRequest openrtb2.BidRequest) ([]*adapters.RequestData, []error) {
+	var requestData []*adapters.RequestData
+	var errors []error
+	networkAdunitMap := make(map[string][]adnAdunit)
+	headers := setHeaders()
+
+	endpoint, err := makeEndpointUrl(ortbRequest, a)
+	if err != nil {
+		errors = append(errors, fmt.Errorf("failed to parse URL: %v", err))
+	}
+
+	for _, imp := range ortbRequest.Imp {
+		if imp.Banner == nil {
+			return nil, []error{&errortypes.BadInput{
+				Message: fmt.Sprintf("ignoring imp id=%s, Adnuntius supports only Banner", imp.ID),
+			}}
+		}
+		var bidderExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			return nil, []error{&errortypes.BadInput{
+				Message: fmt.Sprintf("Error unmarshalling ExtImpBidder: %s", err.Error()),
+			}}
+		}
+
+		var adnuntiusExt openrtb_ext.ImpExtAdnunitus
+		if err := json.Unmarshal(bidderExt.Bidder, &adnuntiusExt); err != nil {
+			return nil, []error{&errortypes.BadInput{
+				Message: fmt.Sprintf("Error unmarshalling ExtImpBmtm: %s", err.Error()),
+			}}
+		}
+
+		network := defaultNetwork
+		if adnuntiusExt.Network != "" {
+			network = adnuntiusExt.Network
+		}
+
+		networkAdunitMap[network] = append(
+			networkAdunitMap[network],
+			adnAdunit{
+				AuId:     adnuntiusExt.Auid,
+				TargetId: fmt.Sprintf("%s-%s", adnuntiusExt.Auid, imp.ID),
+			})
+	}
+
+	site := defaultSite
+	if ortbRequest.Site != nil && ortbRequest.Site.Page != "" {
+		site = ortbRequest.Site.Page
+	}
+
+	for _, networkAdunits := range networkAdunitMap {
+
+		adnuntiusRequest := adnRequest{
+			AdUnits: networkAdunits,
+			Context: site,
+		}
+
+		ortbUser := ortbRequest.User
+		if ortbUser != nil {
+			ortbUserId := ortbRequest.User.ID
+			if ortbUserId != "" {
+				adnuntiusRequest.MetaData.Usi = ortbRequest.User.ID
+			}
+		}
+
+		adnJson, err := json.Marshal(adnuntiusRequest)
+		if err != nil {
+			errors = append(errors, err)
+		}
+
+		if len(errors) > 0 {
+			return nil, errors
+		}
+		requestData = append(requestData, &adapters.RequestData{
+			Method:  http.MethodPost,
+			Uri:     endpoint,
+			Body:    adnJson,
+			Headers: headers,
+		})
+
+	}
+
+	return requestData, nil
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Status code: %d, Request malformed", response.StatusCode),
+		}}
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadInput{
```

Comment:
Can you change this one to `errortypes.BadServerResponse` please? `BadInput` is typically reserved for invalid input but this catch all should typically handle other types of unexpected errors.